db/active_replicator_checkpointer.go

```diff
@@ -295,9 +299,8 @@ func (c *Checkpointer) _updateCheckpointLists() (safeSeq *SequenceID) {
 		}
 	}
 
-	c.stats.ExpectedSequenceLenPostCleanup = len(c.expectedSeqs)
-	c.stats.ProcessedSequenceLenPostCleanup = len(c.processedSeqs)
-
+	c.stats.ExpectedSequenceLenPostCleanup.Set(int64(len(c.processedSeqs)))
```

Comment:
Typo here causing the wrong stat to be set.

```suggestion
	c.stats.ProcessedSequenceLenPostCleanup.Set(int64(len(c.processedSeqs)))
```

This suggests we don't have sufficient test coverage of the stats inside `TestActiveReplicatorPullSkippedSequence` to catch this error. I would recommend these assertions to cover them in the more comprehensive unit test:

```diff
diff --git a/db/active_replicator_checkpointer_test.go b/db/active_replicator_checkpointer_test.go
index 915c9076..e2cb9b82 100644
--- a/db/active_replicator_checkpointer_test.go
+++ b/db/active_replicator_checkpointer_test.go
@@ -267,6 +267,9 @@ func TestCheckpointerSafeSeq(t *testing.T) {
 			})

 			t.Run("_updateCheckpointLists", func(t *testing.T) {
+				originalExpectedSeqsLen := len(tt.c.expectedSeqs)
+				originalProcessedSeqsLen := len(tt.c.processedSeqs)
+
 				actualSafeSeq := tt.c._updateCheckpointLists()
 				if tt.expectedSafeSeq == nil {
 					assert.Nil(t, actualSafeSeq)
@@ -275,6 +278,10 @@ func TestCheckpointerSafeSeq(t *testing.T) {
 				}
 				assert.Equal(t, tt.expectedExpectedSeqs, tt.c.expectedSeqs)
 				assert.Equal(t, tt.expectedProcessedSeqs, tt.c.processedSeqs)
+				assert.Equal(t, int64(originalExpectedSeqsLen), tt.c.stats.ExpectedSequenceLen.Value())
+				assert.Equal(t, int64(len(tt.expectedExpectedSeqs)), tt.c.stats.ExpectedSequenceLenPostCleanup.Value())
+				assert.Equal(t, int64(originalProcessedSeqsLen), tt.c.stats.ProcessedSequenceLen.Value())
+				assert.Equal(t, int64(len(tt.expectedProcessedSeqs)), tt.c.stats.ProcessedSequenceLenPostCleanup.Value())

 				// _updateCheckpointLists should be idempotent
 				// we'd expect no safe seq, and no further changes to either list, as long as c.processedSeqs hasn't been added to
@@ -282,6 +289,8 @@ func TestCheckpointerSafeSeq(t *testing.T) {
 				assert.Nil(t, actualSafeSeq2)
 				assert.Equal(t, tt.expectedExpectedSeqs, tt.c.expectedSeqs)
 				assert.Equal(t, tt.expectedProcessedSeqs, tt.c.processedSeqs)
+				assert.Equal(t, tt.c.stats.ExpectedSequenceLenPostCleanup.Value(), tt.c.stats.ExpectedSequenceLen.Value())
+				assert.Equal(t, tt.c.stats.ProcessedSequenceLenPostCleanup.Value(), tt.c.stats.ProcessedSequenceLen.Value())
 			})
 		})
 	}
```